api/PROFILE.md

```diff
@@ -99,6 +99,56 @@ The request MUST have the body of a single protobuf message of type [LocalAppInf
 Device publishes information repeatedly because of no expectation of persist storage inside local server. Device MAY
 throttle or cancel sending in case of Not implemented response.
 
+### Application Commands
+
+Retrieve a locally submitted commands targeted for application instances running
+on the device.
+
+GET /api/v1/app_commands
+
+Return codes:
+
+* Valid: `200`
+* Not implemented: `404`
+
+The request MUST use HTTP protocol.
+The request MUST NOT contain any body content.
+
+Response:
+
+The response mime type MUST be `application/x-proto-binary`. The response MUST contain
+a single protobuf message of type [LocalAppCmds](./proto/profile/local_profile.proto).
+
+The requester MUST verify that the response payload has the correct `server_token`.
+If the verification succeeds, all entries of `app_commands` are iterated, and those
+that successfully match a running application instance (by `uuid` and/or `displayname`)
+are applied.
+
+Currently, the method allows to request a locally running application instance
+to be *restarted* or *purged* by EVE. This may help to resolve a case of an application
+being in a broken state, and the user not being able to fix it (remotely) due to a lack
+of connectivity between the device and the controller. Rather than rebooting the entire
+device (locally), it is possible to restart/purge only a selected application.
+
+A command request, as defined by `AppCommand` protobuf message, includes an important
+field `timestamp` (`uint64`), which should record the time when the request was made
+by the user. The format of the timestamp is not defined. It can be a Unix timestamp
+or a different time representation. It is not even required for the timestamp to match
+the real time or to be in-sync with the device clock.
+
+What is required, however, is that two successive but distinct requests made for the same
+application and with the same command will have different timestamps attached.
+This requirement applies even between restarts of the Local profile server. A request made
+after a restart should not have the same timestamp attached as the previous request with
+the same arguments made before the restart.
+
+EVE guarantees that a newly added command request (into `LocalAppCmds.app_commands`),
+or a change of the `timestamp` field, will result in the command being triggered ASAP,
+unless the Local Profile Server restarted or the device rebooted shortly after the request
+was made, in which case it could get lost before EVE could see/apply it. This is accepted
+because it is expected that after a restart of the Local profile server, the user would
+repeat any requests that previously failed or didn't yield the desired outcome.
+
```

Comment:
I don't know if there is a feedback channel by which the local profile server can see that EVE has acted on X/Y. If so it would make sense to document it here.

Reply:
We can add this to appinfo (and merge this API with that one), but given the use-case, I would assume that the person in the location has some output from the application that is being restarted/purged. Otherwise how would he know that the command is needed in the first place?

Reply:
I think it needs to be integratable with such a feedback channel otherwise it will be up to the user to manually clear the request based on the user observing that the action has completed.

Reply:
Merged with appinfo and now the feedback is inside info message.

Reply:
Great.
I wonder if the LocalAppInfo should also have an echo timestamp field with the most recently completed app command.
In the case of a purge there is typically a state change to Purging and back to Online, but for a restart this is not necessarily the case.

Reply:
Ah - you have that in the proto already. Might make sense mentioning it here in the doc.

Reply:
done